packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -76,7 +76,31 @@ class PageController extends ScrollController {
   final double viewportFraction;
 
   /// The current page displayed in the controlled [PageView].
+  ///
+  /// There are circumstances that this [PageController] can't know the current
+  /// page. Mainly when no [PageView]s are attached. Reading [page] will throw
+  /// an [AssertionError] in the following cases:
+  ///
+  /// 1. Zero [PageView] attached. On attaching or re-attaching, the new [page]
+  /// position will be derived:
+  ///
+  ///   * First, based on the attached [PageView]'s [BuildContext] and the
+  ///     position saved at that context's [PageStorage] if [keepPage] is true.
+  ///   * Second, from the [PageController]'s [initialPage].
+  ///
+  /// 2. More than one [PageView] attached to the same [PageController].
+  ///
+  /// [hasClients] can be used to check if a [PageView] is attached prior to
+  /// accessing [page].
   double get page {
+    assert(
+      positions.isNotEmpty,
+      'PageController.page cannot be accessed before a PageView is built with it',
+    );
+    assert(
+      positions.length == 1,
+      'Multiple PageViews cannot be attached to the same PageController',
+    );
```

Comment:
these assertion messages should end with a period.

Reply:
Done